src/operations/blas3/gemm_local_tensorcore.hpp

```diff
@@ -0,0 +1,814 @@
+      const index_t &n, const index_t &orig_k, const index_t &mc,
+      const index_t &nc, const index_t &a_size, const index_t &b_size,
+      const index_t &c_size, InputPointerType orig_A, const index_t &lda,
+      InputPointerType orig_B, const index_t &ldb, OutputPointerType orig_C,
+      const index_t &ldc, OutputScratchPointerType s0,
+      InputScratchPointerType s1, InputScratchPointerType s2,
+      InputScratchPointerType s3, InputScratchPointerType s4,
+      const bool out_of_range, index_t batch_stride, index_t wg_batch_id,
+      index_t batch_size) noexcept {
+    index_t ofs = 1;
+    using namespace cl::sycl::ext::oneapi::experimental::matrix;
+    using CType = joint_matrix<typename tile_type::jmOutType,
+                               matrix_use::accumulator, tile_type::sg_M,
+                               tile_type::sg_N, matrix_layout::col_major>;
+    do {
+      auto A = orig_A;
+      auto B = orig_B;
+      auto C = orig_C;
+      auto k = orig_k;
+      CType reg_res[frags_per_sg] = {};
+      while (k >= cl_elems) {
+        extract_input_blocks<check_m_limit, check_n_limit, false>(
+            item_id, m, n, k, A, lda, B, ldb, s1, s3, out_of_range);
+        id.barrier(cl::sycl::access::fence_space::local_space);
+        compute_block_gemm<check_m_limit, check_n_limit>(id, s2, s4, reg_res);
+        A += cl_elems * (trans_a ? 1 : lda);
+        B += cl_elems * (trans_b ? ldb : 1);
+
+        sync_smem<double_buffer, block_cols * ldsb, block_cols * ldsb,
+                  ldsa * cl_elems, ldsa * cl_elems>(id, ofs, s1, s2, s3, s4);
+        k -= cl_elems;
+      }
+
+      if (k > 0) {
+        extract_input_blocks<check_m_limit, check_n_limit, true>(
+            item_id, m, n, k, A, lda, B, ldb, s1, s3, out_of_range);
+        id.barrier(cl::sycl::access::fence_space::local_space);
+        compute_block_gemm<check_m_limit, check_n_limit>(id, s2, s4, reg_res);
+
+        sync_smem<double_buffer, block_cols * ldsb, block_cols * ldsb,
+                  ldsa * cl_elems, ldsa * cl_elems>(id, ofs, s1, s2, s3, s4);
+      }
+
+      // store the output
+      store_output_block<check_m_limit, check_n_limit>(id, mc, nc, C, s0, ldc,
+                                                       reg_res, out_of_range);
+      orig_A += (a_size * batch_stride);
+      orig_B += (b_size * batch_stride);
+      orig_C += (c_size * batch_stride);
+      // batch_size_ must be signed as the negative value has meaning here.
+      batch_size -= batch_stride;
+    } while (batch_size > wg_batch_id);
+  }
+
+  /*!
+   * @brief Store the computed gemm result to the C matrix after performing
+   * input and output scaling. Output scaling only takes place if beta is
+   * non-zero.
+   *
+   * @tparam check_m_limit  iff true, check if row indexes of C are
+   *                        out-of-bound
+   * @tparam check_n_limit  iff true, check if no indexes of C are
+   *                        out-of-bound
```

Comment:
When referring to a matrix, I think this should be indices - eg see [wikipedia](https://en.wikipedia.org/wiki/Index_notation). 

This comment also applies to use of indexes elsewhere.

Reply:
Appears to have been applied.